docs/assignment3.md

```diff
@@ -1 +1,577 @@
+
+Looking torwards the right hand side of the diagram, or the **PlayerThread** side, we see the 2 players early mentioned. Each **Player** is controlled by an individual *Thread*, instantiated on the **PlayerThread** object. Each thread must have been instatiated with a *Socket*, which will be used to instantiate the **Reader** and **Writer** objects. The **Reader** objects hold the *inputStream* while the **Writer** holds the *outputStream*. Looking at 1 Player specifically, Jack, we see he is currently in the **Area** 0,1 of the **Map**. To better understand how an the *Instructions* work, pay attention to the **Instruction** object associated with *Jack's* **Parser**. To get this object, the *String* inputted must have been "Lift carpet", yield an **Instruction** with "lift" as an *action*, and {"carpet",null,null} as the *items*. The **PlayerThread** will recived this object when calling **Player.getInstruction**(), and pass it to the *game.validateInstruction*(<instruction just recieved>, <player the thread controls>). The **Game** will then decide, using the program logic, wether the move is valid, and if so, adjust the attributes of **Player** (Jack) appropriatly. *Mark* is very similar to *Jack*, except *Mark* was instantiated using its appropriate *Socket*, and *Mark* has a *Broom* and *Sword* in his *backpack*. 
+
+>ClientSide Object Diagram (Diagram 4)
+
+![ClientSide Object Diagram](https://i.postimg.cc/c4zhjvVC/Class-diagram-Object-Diagram-2-1.png)
+
+In the ServerSide, we saw the **Player** recieving input *Strings* from the **Parser**, which held a connection to *Socket*. The otherside of a *Socket* belongs to the system above, a *ClientSide* System. This system is simply a multiThreaded chat-cleint to the ServerSide. When a user inputs a *String* to send to the *Server*, the Server will recieve this *String*, process it when ready (*game.isbusy* == false). Then, the Server will handle the request, and ALWAYS reply with a *String* indicating failure/success of the action taken. The *Server* may also reply with more verbose information such as **Area** description etc. The asynchronus (multithreaded) approach of the **playerClient** allows the *Server* to send information to the *Client*, without the *Client* have sent a request first. This is especially useful when inter-player interaction takes place. An example of inter-player interaction is talking, or trading. 
+
+The Interface is also linked to the read and writer threads. This is needed to direct I/O to the correct panels for the interface. Why we decided to use this interace is better described in <a href="whyInterface">ClassDiagram.Interface</a>. Below is an image of the **Interface** we created: 
+
+<img src="https://i.postimg.cc/vHbZ16dK/Screen-Shot-2020-03-30-at-9-35-39-AM.png" alt="Image of Client Interface" style="zoom:30%;" />
+
+## State machine diagrams									
+Author(s): `Irene Garcia-Fortea Garcia, Florent Brunet de Rochebrune`
```

Comment:
In the execute command state, the game.isBusy assignment should have only one '=' . So game.isBusy = True